### DISCUSSION TITLE
Incompatible mod set

### DISCUSSION THREAD

**Discussion by NotEonzz** [FabricMC/fabric-api#2379] [Player support]
I was just trying to start my server, and got this error message in the console.

05.07 04:28:22 [Server] INFO WARNING: Published ports are discarded when using host network mode
05.07 04:28:22 [Server] INFO 6e4240e7b2fae4db76f4d1808c670bab08698c7e39a1293ade98b0bc501410ca
05.07 04:28:24 [Server] INFO Starting net.fabricmc.loader.impl.game.minecraft.BundlerClassPathCapture
05.07 04:28:24 [Server] INFO [Log4jPatcher] [INFO] Transforming org/apache/logging/log4j/core/pattern/MessagePatternConverter
05.07 04:28:24 [Server] INFO [Log4jPatcher] [WARN]  Unable to find noLookups:Z field in org/apache/logging/log4j/core/pattern/MessagePatternConverter
05.07 04:28:24 [Server] INFO [Log4jPatcher] [INFO] Transforming org/apache/logging/log4j/core/lookup/JndiLookup
05.07 04:28:24 [Server] main/INFO Loading Minecraft 1.19 with Fabric Loader 0.14.8
05.07 04:28:24 [Server] main/ERROR Incompatible mod set!
05.07 04:28:24 [Server] INFO net.fabricmc.loader.impl.FormattedException: net.fabricmc.loader.impl.discovery.ModResolutionException: Mod discovery failed!
05.07 04:28:24 [Server] INFO at net.fabricmc.loader.impl.FabricLoaderImpl.load(FabricLoaderImpl.java:192) ~[fabric-loader-0.14.8.jar:?]
05.07 04:28:24 [Server] INFO at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:148) ~[fabric-loader-0.14.8.jar:?]
05.07 04:28:24 [Server] INFO at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:68) [fabric-loader-0.14.8.jar:?]
05.07 04:28:24 [Server] INFO at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) [fabric-loader-0.14.8.jar:?]
05.07 04:28:24 [Server] INFO at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69) [fabric-loader-0.14.8.jar:?]
05.07 04:28:24 [Server] INFO Caused by: net.fabricmc.loader.impl.discovery.ModResolutionException: Mod discovery failed!
05.07 04:28:24 [Server] INFO at net.fabricmc.loader.impl.discovery.ModDiscoverer.lambda$discoverMods$1(ModDiscoverer.java:146) ~[fabric-loader-0.14.8.jar:?]
05.07 04:28:24 [Server] INFO at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33) ~[fabric-loader-0.14.8.jar:?]
05.07 04:28:24 [Server] INFO at net.fabricmc.loader.impl.discovery.ModDiscoverer.discoverMods(ModDiscoverer.java:146) ~[fabric-loader-0.14.8.jar:?]
05.07 04:28:24 [Server] INFO at net.fabricmc.loader.impl.FabricLoaderImpl.setup(FabricLoaderImpl.java:210) ~[fabric-loader-0.14.8.jar:?]
05.07 04:28:24 [Server] INFO at net.fabricmc.loader.impl.FabricLoaderImpl.load(FabricLoaderImpl.java:187) ~[fabric-loader-0.14.8.jar:?]
05.07 04:28:24 [Server] INFO ... 4 more
05.07 04:28:24 [Server] INFO Caused by: java.lang.RuntimeException: java.lang.RuntimeException: Error analyzing [/mods/better-end-2.0.5.jar]: java.util.zip.ZipException: zip END header not found
05.07 04:28:24 [Server] INFO at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
05.07 04:28:24 [Server] INFO at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77) ~[?:?]
05.07 04:28:24 [Server] INFO at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
05.07 04:28:24 [Server] INFO at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499) ~[?:?]
05.07 04:28:24 [Server] INFO at java.lang.reflect.Constructor.newInstance(Constructor.java:480) ~[?:?]
05.07 04:28:24 [Server] INFO at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:562) ~[?:?]
05.07 04:28:24 [Server] INFO at java.util.concurrent.ForkJoinTask.reportExecutionException(ForkJoinTask.java:604) ~[?:?]
05.07 04:28:24 [Server] INFO at java.util.concurrent.ForkJoinTask.get(ForkJoinTask.java:981) ~[?:?]
05.07 04:28:24 [Server] INFO at net.fabricmc.loader.impl.discovery.ModDiscoverer.discoverMods(ModDiscoverer.java:143) ~[fabric-loader-0.14.8.jar:?]
05.07 04:28:24 [Server] INFO at net.fabricmc.loader.impl.FabricLoaderImpl.setup(FabricLoaderImpl.java:210) ~[fabric-loader-0.14.8.jar:?]
05.07 04:28:24 [Server] INFO at net.fabricmc.loader.impl.FabricLoaderImpl.load(FabricLoaderImpl.java:187) ~[fabric-loader-0.14.8.jar:?]
05.07 04:28:24 [Server] INFO ... 4 more
05.07 04:28:24 [Server] INFO Caused by: java.lang.RuntimeException: Error analyzing [/mods/better-end-2.0.5.jar]: java.util.zip.ZipException: zip END header not found
05.07 04:28:24 [Server] INFO at net.fabricmc.loader.impl.discovery.ModDiscoverer$ModScanTask.compute(ModDiscoverer.java:255) ~[fabric-loader-0.14.8.jar:?]
05.07 04:28:24 [Server] INFO at net.fabricmc.loader.impl.discovery.ModDiscoverer$ModScanTask.compute(ModDiscoverer.java:212) ~[fabric-loader-0.14.8.jar:?]
05.07 04:28:24 [Server] INFO at java.util.concurrent.RecursiveTask.exec(RecursiveTask.java:100) ~[?:?]
05.07 04:28:24 [Server] INFO at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[?:?]
05.07 04:28:24 [Server] INFO at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[?:?]
05.07 04:28:24 [Server] INFO at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[?:?]
05.07 04:28:24 [Server] INFO at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[?:?]
05.07 04:28:24 [Server] INFO at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[?:?]
05.07 04:28:24 [Server] INFO Caused by: java.util.zip.ZipException: zip END header not found
05.07 04:28:24 [Server] INFO at java.util.zip.ZipFile$Source.findEND(ZipFile.java:1469) ~[?:?]
05.07 04:28:24 [Server] INFO at java.util.zip.ZipFile$Source.initCEN(ZipFile.java:1477) ~[?:?]
05.07 04:28:24 [Server] INFO at java.util.zip.ZipFile$Source.<init>(ZipFile.java:1315) ~[?:?]
05.07 04:28:24 [Server] INFO at java.util.zip.ZipFile$Source.get(ZipFile.java:1277) ~[?:?]
05.07 04:28:24 [Server] INFO at java.util.zip.ZipFile$CleanableResource.<init>(ZipFile.java:709) ~[?:?]
05.07 04:28:24 [Server] INFO at java.util.zip.ZipFile.<init>(ZipFile.java:243) ~[?:?]
05.07 04:28:24 [Server] INFO at java.util.zip.ZipFile.<init>(ZipFile.java:172) ~[?:?]
05.07 04:28:24 [Server] INFO at java.util.zip.ZipFile.<init>(ZipFile.java:186) ~[?:?]
05.07 04:28:24 [Server] INFO at net.fabricmc.loader.impl.discovery.ModDiscoverer$ModScanTask.computeJarFile(ModDiscoverer.java:280) ~[fabric-loader-0.14.8.jar:?]
05.07 04:28:24 [Server] INFO at net.fabricmc.loader.impl.discovery.ModDiscoverer$ModScanTask.compute(ModDiscoverer.java:250) ~[fabric-loader-0.14.8.jar:?]
05.07 04:28:24 [Server] INFO at net.fabricmc.loader.impl.discovery.ModDiscoverer$ModScanTask.compute(ModDiscoverer.java:212) ~[fabric-loader-0.14.8.jar:?]
05.07 04:28:24 [Server] INFO at java.util.concurrent.RecursiveTask.exec(RecursiveTask.java:100) ~[?:?]
05.07 04:28:24 [Server] INFO at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[?:?]
05.07 04:28:24 [Server] INFO at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[?:?]
05.07 04:28:24 [Server] INFO at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[?:?]
05.07 04:28:24 [Server] INFO at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[?:?]
05.07 04:28:24 [Server] INFO at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[?:?]

**Comment by Daomephsta** [comment]
> Error analyzing [/mods/better-end-2.0.5.jar]: java.util.zip.ZipException: zip END header not found
 
` better-end-2.0.5.jar` is corrupt and cannot be read. Download and install it again.
